finality-aleph/src/data_io.rs

```diff
@@ -25,8 +27,31 @@ const AVAILABLE_BLOCKS_CACHE_SIZE: usize = 1000;
 const MESSAGE_ID_BOUNDARY: MessageId = 100_000;
 const PERIODIC_MAINTENANCE_INTERVAL: Duration = Duration::from_millis(60000);
 
+#[derive(Clone, Debug, Encode, Decode)]
+pub(crate) struct AlephData<B: BlockT> {
+    pub hash: B::Hash,
+    pub number: NumberFor<B>,
+}
+
+impl<B: BlockT> Hash for AlephData<B> {
```

Comment:
Unfortunately when I use derive here it says that `B` should also implement `Hash`. This way it doesn't

Reply:
I have tried it, and it doesn't work indeed. I think we should solve it somehow ;/. The suggestion that B should implement Hash doesn't make sense at all and makes me think that either we do something wrong, or there is a problem with the compiler.

Reply:
I agree, but this is the only workaround I have found.

Reply:
You could make this struct be parametrized by the values directly, instead of by the whole `B` block type. Then you could add an alias that would only use the block, for convenience.